### PR TITLE
Clean up prereq eval code

### PR DESCRIPTION
**Pull request by drichelson**
Clean up prereq eval code

### CODE REVIEW

File: src/main/java/com/launchdarkly/client/FeatureFlag.java [hunk 1]
```diff
@@ -175,12 +174,10 @@ VariationOrRollout getFallthrough() {
   static class EvalResult {
     private JsonElement value;
     private List<FeatureRequestEvent> prerequisiteEvents;
-    private Set<String> visitedFeatureKeys;
 
-    private EvalResult(JsonElement value, List<FeatureRequestEvent> prerequisiteEvents, Set<String> visitedFeatureKeys) {
+    private EvalResult(JsonElement value, List<FeatureRequestEvent> prerequisiteEvents) {
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: I don't even think this class is necessary anymore. Can't you just collect the events via the list passed as the function argument?

Comment (drichelson) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: For the internal method yes, but we still need to return 2 objects (value + prereq events)
I removed the use of the EvalResult class from the internal method- hopefully that makes things easier.

File: src/main/java/com/launchdarkly/client/FeatureFlag.java [hunk 2]
```diff
@@ -67,40 +67,39 @@ EvalResult evaluate(LDUser user, FeatureStore featureStore) {
...
-  // Returning either a nil EvalResult or EvalResult.value indicates prereq failure/error.
-  private EvalResult evaluate(LDUser user, FeatureStore featureStore, List<FeatureRequestEvent> events, Set<String> visited) {
+  // Returning either a JsonElement or null indicating prereq failure/error.
+  private JsonElement evaluate(LDUser user, FeatureStore featureStore, List<FeatureRequestEvent> events, Set<String> visited) {
     boolean prereqOk = true;
-    EvalResult evalResult = new EvalResult(null, events, visited);
     for (Prerequisite prereq : prerequisites) {
-      evalResult.visitedFeatureKeys.add(key);
-      if (evalResult.visitedFeatureKeys.contains(prereq.getKey())) {
+      visited.add(key);
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: This line is in the wrong place. You're adding the feature flag's key with each iteration over the prerequisites.

**Comment by jkodumal** [comment]
Once that last comment is addressed this is 👍 

It looks pretty similar to what I wrote in the node client (modulo the callback hell), which is encouraging.

**Comment by jkodumal** [comment]
👍

**Pull request merged**